sdk/src/test/java/com/mapbox/maps/renderer/FpsManagerTest.kt

```diff
@@ -0,0 +1,196 @@
+package com.mapbox.maps.renderer
+
+import android.os.Handler
+import android.os.Looper
+import android.view.Choreographer
+import com.mapbox.maps.logI
+import com.mapbox.maps.logW
+import com.mapbox.maps.renderer.FpsManager.Companion.VSYNC_COUNT_TILL_IDLE
+import io.mockk.*
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.Shadows
+import org.robolectric.annotation.LooperMode
+import org.robolectric.shadows.ShadowChoreographer
+import java.util.concurrent.TimeUnit
+
+@RunWith(RobolectricTestRunner::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class FpsManagerTest {
+
+  private lateinit var fpsManager: FpsManager
+
+  @Before
+  fun setUp() {
+    fpsManager = FpsManager(Handler(Looper.getMainLooper()))
+    mockkStatic("com.mapbox.maps.MapboxLogger")
+    every { logI(any(), any()) } just Runs
+    every { logW(any(), any()) } just Runs
+    // simulate all the tests on 60 Hz screen, meaning we have 16.6 ms time to render a frame
+    ShadowChoreographer.setPostFrameCallbackDelay(CHOREOGRAPHER_POST_FRAME_CALLBACK_DELAY_MS)
+    fpsManager.setScreenRefreshRate(SCREEN_FPS)
+  }
+
+  @Test
+  fun userRefreshRateNotSetTest() {
+    pauseHandler()
+    idleHandler(vsyncCount = 10) {
+      assert(fpsManager.preRender(it))
+    }
+  }
+
+  @Test
+  fun userRefreshRate24Test() {
+    fpsManager.setUserRefreshRate(24)
+    pauseHandler()
+    val frameRenderedPattern = mutableListOf<Boolean>()
+    idleHandler(vsyncCount = 10) {
+      frameRenderedPattern.add(fpsManager.preRender(it))
+    }
+    Assert.assertArrayEquals(
+      arrayOf(
+        false, false, true, false, true,
+        false, false, true, false, true
+      ),
+      frameRenderedPattern.toTypedArray()
+    )
+  }
+
+  @Test
+  fun userRefreshRate7Test() {
+    fpsManager.setUserRefreshRate(7)
+    pauseHandler()
+    val frameRenderedPattern = mutableListOf<Boolean>()
+    idleHandler(vsyncCount = 18) {
+      frameRenderedPattern.add(fpsManager.preRender(it))
+    }
+    Assert.assertArrayEquals(
+      arrayOf(
```

Comment:
`Array(vsyncCount) { it == 9 || it == 18 }`

Reply:
Hmm, and why 9/18 if FPS is 7?

Reply:
Overall I would write those tests to be at least 1 or 2 seconds, e.g. use vsyncCount = 120 everywhere

Reply:
> Hmm, and why 9/18 if FPS is 7?

A bit of math - to achieve 7 FPS on 60 Hz screen we have following ratio: 7 / 60 = 0.11(6) => thus we render new frame roughly each 9th frame. I don't see much sense in adding bigger vsync interval but just for you I'll add one additional test with some strange rate and with range of 120 frames.